cmd/ocsp-updater/main.go

```diff
@@ -132,22 +148,39 @@ func newUpdater(
 		batchSize:                    config.OldOCSPBatchSize,
 		maxBackoff:                   config.SignFailureBackoffMax.Duration,
 		backoffFactor:                config.SignFailureBackoffFactor,
+		serialSuffixes:               serialSuffixes,
 	}
 
 	return &updater, nil
 }
 
+func (updater *OCSPUpdater) getQuestionsForShardList() string {
+	return strings.TrimRight(strings.Repeat("?,", len(updater.serialSuffixes)), ",")
+}
+
+func (updater *OCSPUpdater) appendShardList(params []interface{}) []interface{} {
+	for _, c := range updater.serialSuffixes {
+		params = append(params, c)
+	}
+	return params
+}
+
 func (updater *OCSPUpdater) findStaleOCSPResponses(oldestLastUpdatedTime time.Time, batchSize int) ([]core.CertificateStatus, error) {
-	statuses, err := sa.SelectCertificateStatuses(
-		updater.readOnlyDbMap,
-		`WHERE ocspLastUpdated < :lastUpdate
+	queryBody := fmt.Sprintf(`WHERE ocspLastUpdated < ?
 		 AND NOT isExpired
+		 AND RIGHT(serial, 1) IN ( %s )
 		 ORDER BY ocspLastUpdated ASC
-		 LIMIT :limit`,
-		map[string]interface{}{
-			"lastUpdate": oldestLastUpdatedTime,
-			"limit":      batchSize,
-		},
+		 LIMIT ?`, updater.getQuestionsForShardList())
+
+	params := make([]interface{}, 0)
+	params = append(params, oldestLastUpdatedTime)
+	params = updater.appendShardList(params)
```

Comment:
You can collapse the logic from method `appendShardList` into this part of `findStaleOCSPResponses` and remove the tests for it. The tests are only checking that `append` works. The thing we would care about is that we're not just appending empty slice, but you already check that on line `94`.

Reply:
Yes, exclelent.